test/distributed/pipelining/test_schedule_multiproc.py

```diff
@@ -383,6 +383,70 @@ def test_grad_with_manual_interleaved(self, ScheduleClass):
                     print(f"Gradient test failed for {name}: {p.grad} vs {ref_p.grad}")
                     raise
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    @parametrize("ScheduleClass", [ScheduleWithW])
+    def test_schedule_with_native_zero_bubble(self, ScheduleClass):
```

Comment:
looks like this is missing output/grad check, any reason not to just use the existing grad test?

Reply:
Finally fixed the grad bug and added a grad check.

I will clean up schedule args and add a new schedule specifically for zero-bubble to consolidate the tests in a followup PR

Reply:
my question is whether there is a reason to duplicate the code of the test.  Is it identical to the other test_schedule that is already parametrized?  (should we just delete this test, and add `[ScheduleWithW, ScheduleFlexibleInterleaved1F1B]` to the other test, or is it useful to separate them?)

Reply:
Yeah the test has a lot of duplicate code, but also the schedule args are different so i would have to add if-statements in the `test_grad_with_manual_interleaved`. I think the test can be deleted but and added as one the parameterized tests, but I will do it as cleanup after I add the official zero bubble schedule